cmds/bmp/create_baremetals_command.go

```diff
@@ -45,10 +52,57 @@ func (cmd createBaremetalsCommand) Options() cmds.Options {
 
 func (cmd createBaremetalsCommand) Validate() (bool, error) {
 	cmd.printer.Printf("Validating %s command: options: %#v", cmd.Name(), cmd.options)
+	if cmd.options.Deployment == "" {
+		return false, errors.New("please specify the deployment file with -d")
+	}
+
+	_, err := os.Stat(cmd.options.Deployment)
+	if os.IsNotExist(err) {
+		return false, errors.New(fmt.Sprintf("deployment file %s doesn't exist", cmd.options.Deployment))
+	}
+
 	return true, nil
 }
 
 func (cmd createBaremetalsCommand) Execute(args []string) (int, error) {
 	cmd.printer.Printf("Executing %s comamnd: args: %#v, options: %#v", cmd.Name(), args, cmd.options)
+
+	filename, _ := filepath.Abs(cmd.options.Deployment)
+	yamlFile, err := ioutil.ReadFile(filename)
+
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: could not read File %s, error message %s", filename, err.Error())
+		return 1, errors.New(errorMessage)
+	}
+
+	var deploymentInfo clients.Deployment
+	err = yaml.Unmarshal(yamlFile, &deploymentInfo)
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: failed to decode Yaml File %s, error message %s", filename,
+			err.Error())
+		return 1, errors.New(errorMessage)
+	}
+
+	createBaremetalsInfo := clients.CreateBaremetalsInfo{}
+	createBaremetalsInfo.Deployment = deploymentInfo.Name
+	for _, resource := range deploymentInfo.ResourcePools {
+		if resource.CloudProperty.Baremetal {
+			createBaremetalsInfo.BaremetalSpecs = append(createBaremetalsInfo.BaremetalSpecs,
+				resource.CloudProperty)
+		}
+	}
+
+	createBaremetalsResponse, err := cmd.bmpClient.CreateBaremetals(createBaremetalsInfo, cmd.options.DryRun)
+	if err != nil {
+		return 1, err
+	}
+
+	if createBaremetalsResponse.Status != 200 {
+		return createBaremetalsResponse.Status, nil
+	}
+
+	taskId := createBaremetalsResponse.Data.TaskId
```

Comment:
If you are not using this except in the `cmd.ui.Printf` call, I suggest you don't create the variable, so something like:

```
cmd.ui.Printf("Run command: bmp task %d to get the status of the task", createBaremetalsResponse.Data.TaskId)
```